third_party/xla/third_party/tsl/tsl/platform/threadpool.cc

```diff
@@ -107,6 +107,12 @@ ThreadPool::ThreadPool(Env* env, const ThreadOptions& thread_options,
                        bool low_latency_hint, Eigen::Allocator* allocator) {
   CHECK_GE(num_threads, 1);
 
+#ifdef DNNL_AARCH64_USE_ACL
+  if(num_threads == std::thread::hardware_concurrency() && num_threads >= 16){
```

Comment:
Could you please add the reason as a comment in this code?

Reply:
Thanks for the review! I have now changed to use `tsl::port::NumTotalCPUs()` to fetch the number of cores.

Regarding the cause this patch addresses a use case when any process (not only TF) is allocated to use all cores on machine that has other services that are running. We have observed that on some systems when process is allocated to use all available cores one of the threads of the process get swapped out by Linux’s completely fair scheduler so that other processes can make some progress returning core to the thread of the process that got swapped out, which causes performance issues. If we use schedulers where threads of the process can busy wait and therefore not allow other process to proceed we do not see this issue.

Reply:
If you're mainly talking about overall core counts, then it would affect systems with num_threads < 16 as well.  So where does the large number of cores come into play?